TELCODOCS-1975 - ZTP Configuring the hub cluster for backup and restore (#88229)
openshift/openshift-docs: modules/ztp-configuring-the-hub-cluster-for-backup-and-restore.adoc

// Module included in the following assemblies:
//
// * scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc

:_mod-docs-content-type: PROCEDURE
[id="ztp-configuring-the-hub-cluster-for-backup-and-restore_{context}"]
= Configuring the hub cluster for backup and restore

> You can use {ztp} to configure a set of policies to backup `BareMetalHost` resources. 

Reviewer: ```suggestion
You can use {ztp} to configure a set of policies to backup `BareMetalHost` resources. 
```
I know this already went through peer review and this is a cherry pick, but when used as a verb, "back up" [should be two words](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#b).

Author: Thanks for reviewing @kcarmichael08

            {{- end }}
          remediationAction: enforce
          severity: high
---
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: Placement
metadata:
  name: bmh-cluster-activation-label-pr
spec:
  predicates:
    - requiredClusterSelector:
        labelSelector:
          matchExpressions:
            - key: name
              operator: In
              values:
                - local-cluster
---
apiVersion: policy.open-cluster-management.io/v1
kind: PlacementBinding
metadata:
  name: bmh-cluster-activation-label-binding
placementRef:
  name: bmh-cluster-activation-label-pr
  apiGroup: cluster.open-cluster-management.io
  kind: Placement
subjects:
  - name: bmh-cluster-activation-label
    apiGroup: policy.open-cluster-management.io
    kind: Policy
---
apiVersion: cluster.open-cluster-management.io/v1beta2
kind: ManagedClusterSetBinding
metadata:
  name: default
  namespace: default
spec:
  clusterSet: default
----
<1> If you apply the `cluster.open-cluster-management.io/backup: cluster-activation` label to `BareMetalHost` resources, the {rh-rhacm} cluster backs up those resources.

> You can restore the `BareMetalHost` resources if the active cluster becomes unavailable, when restoring the hub activation resources.

Reviewer: ```suggestion
You can restore the `BareMetalHost` resources if the active cluster becomes unavailable when restoring the hub activation resources.
```

Author: I think the comma is correct here. Otherwise it might read as the active cluster becoming unavailable when the user is restoring the resources.

. Apply the policy by running the following command:
+
[source,terminal]
----
$ oc apply -f BareMetalHostBackupPolicy.yaml
----

.Verification

. Find all `BareMetalHost` resources with the label `infraenvs.agent-install.openshift.io`  by running the following command: 
+
[source,terminal]
----
$ oc get BareMetalHost -A -l infraenvs.agent-install.openshift.io
----
+
.Example output
[source,yaml]
----
NAMESPACE      NAME             STATE   CONSUMER   ONLINE   ERROR   AGE
baremetal-ns   baremetal-name                      false            50s
----

> . Verify that the policy has applied the label `cluster.open-cluster-management.io/backup=cluster-activation` to all these resources, by runing the following command:

Reviewer: ```suggestion
. Verify that the policy has applied the label `cluster.open-cluster-management.io/backup=cluster-activation` to all these resources, by running the following command:
```

+
[source,terminal]
----
$ oc get BareMetalHost -A -l infraenvs.agent-install.openshift.io,cluster.open-cluster-management.io/backup=cluster-activation
----
+
.Example output
[source,yaml]
----
NAMESPACE      NAME             STATE   CONSUMER   ONLINE   ERROR   AGE
baremetal-ns   baremetal-name                      false            50s
----
+
The output must show the same list as in the previous step, which listed all `BareMetalHost` resources with the label `infraenvs.agent-install.openshift.io`.
This confirms that that all the `BareMetalHost` resources with the `infraenvs.agent-install.openshift.io` label also have the `cluster.open-cluster-management.io/backup: cluster-activation` label.
+
The following example shows a `BareMetalHost` resource with the `infraenvs.agent-install.openshift.io` label. 

> The resource must also have the `cluster.open-cluster-management.io/backup: cluster-activation` label, which was added by the policy created in step 1.

Reviewer: ```suggestion
The resource must also have the `cluster.open-cluster-management.io/backup: cluster-activation` label, which was added by the policy that you created.
```
"in step 1" is confusing because the closest step 1 to this step is the verification step, not the policy creation step in the preceding section. I think this could just be omitted.

+
[source,yaml]
----
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  labels:
    cluster.open-cluster-management.io/backup: cluster-activation
    infraenvs.agent-install.openshift.io: value
  name: baremetal-name
  namespace: baremetal-ns
----

You can now use {rh-rhacm-title} to restore a managed cluster. 

[IMPORTANT]
====
When you restore `BareMetalHosts` resources as part of restoring the cluster activation data, you must restore the `BareMetalHosts` status.
The following {rh-rhacm} `Restore` resource example restores activation resources, including `BareMetalHosts`, and also restores the status for the `BareMetalHosts` resources:
[source,yaml]
----

>   apiVersion: cluster.open-cluster-management.io/v1beta1

Reviewer: I think this should be left justified? (there are extra spaces in front of `apiVersion`)
